pkg/eventpersister/persister/persister.go

```diff
@@ -614,3 +548,141 @@ func (p *Persister) key(
 		environmentNamespace,
 	)
 }
+
+// validateTimestamp limits date range of the given timestamp
+func (p *Persister) validateTimestamp(
+	timestamp int64,
+	oldestTimestampDuration, furthestTimestampDuration time.Duration,
+) bool {
+	given := time.Unix(timestamp, 0)
+	maxPast := time.Now().Add(-oldestTimestampDuration)
+	if given.Before(maxPast) {
+		return false
+	}
+	maxFuture := time.Now().Add(furthestTimestampDuration)
+	return !given.After(maxFuture)
+}
+
+func (p *Persister) getEvaluation(
+	ctx context.Context,
+	event *eventproto.GoalEvent,
+	environmentNamespace string,
+) (*featureproto.Evaluation, bool, error) {
+	// List experiments with the following status RUNNING, FORCE_STOPPED, and STOPPED
+	experiments, err := p.listExperiments(ctx, environmentNamespace)
+	if err != nil {
+		return nil, true, err
+	}
+	if len(experiments) == 0 {
+		handledCounter.WithLabelValues(codeNoExperiments).Inc()
+		return nil, false, ErrNoExperiments
+	}
+	// Find the experiment by goal ID
+	// TODO: we must change the console UI not to allow creating
+	// multiple experiments running at the same time,
+	// using the same feature flag id and goal id
+	var experiment *exproto.Experiment
```

Comment:
As described in the comment, we need to change the Experiment API to not allow creating experiments using the same feature flag id, version, and goal id. This case is rare, but we should make the changes soon.